jupyterlab_gcpscheduler/src/components/dialog.tsx

```diff
@@ -263,11 +250,20 @@ export class SchedulerDialog extends React.Component<Props, State> {
   }
 
   // Casts to GcpSettings shape from JSONObject
-  private _settingsChanged(newSettings: ISettingRegistry.ISettings) {
+  private async _settingsChanged(newSettings: ISettingRegistry.ISettings) {
     const settings = (newSettings.composite as unknown) as GcpSettings;
     if (settings.projectId) {
       console.info(`Using ${settings.projectId} for GCP API calls`);
       this.props.projectStateService.projectId = settings.projectId;
+    } else {
+      let projectId = undefined;
+      try {
+        projectId = await this.props.gcpService.projectId;
+      } catch (err) {
+        console.info(`${err} occurred when trying to get projectId`);
```

Comment:
I think this is pretty much a fatal condition, albeit one that shouldn't occur in practice. I'm wondering if we should just let it error out which would cause the extension not to load.

Reply:
can do